### ISSUE TITLE
Incompatibility with Xamarin.Android.Support.CustomTabs 28

### ISSUE THREAD

**Issue by taublast** [OneSignal/OneSignal-Xamarin-SDK#127]
**Description:**
Hi we have Xamarin.Forms updated and OneSignal 3.2.2 just doesn't let us use updated nugets.

SDK:
OneSignal 3.2.2 and xamarin 3.4.0.100

Reproduce:

A. Install latest xamarin 3.4.0.100. Now you cannot install OneSignal nuget at all as it's incompatible with latest xamarin. With Xamarin.Android.Support.CustomTabs 28 to start.

OR

B. Install OneSignal nuget. You cannot install latest xamarin as its incompatible with xamarin android nugets as above.

**Comment by bares43** [comment]
Same problem here. Any solution?

**Comment by moreirawebmaster** [comment]
Same problem here. Any solution? ²

**Comment by jabbourac** [comment]
I'm experiencing the same issue as well. It's causing my Android build to crash. Any solution?

**Comment by EmilAlipiev** [comment]
I have reported that wayyy long time a go and they said thats the way it is.

**Comment by ArgusMagnus** [comment]
Same issue

**Comment by snice1981** [comment]
same problem, please provide a solution for that. Thank you

**Comment by jkasten2** [comment]
@taublast @bares43 @moreirawebmaster @jabbourac @lucianoprevedello @EmilAlipiev @ArgusMagnus @snice1981 Sorry for the delay, we just released a 3.3.0-beta version with updated dependencies.
https://www.nuget.org/packages/Com.OneSignal/3.3.0-beta

Please try it out and let us know if it resolves this issue with MonoAndroid 9.0 compatibility. Thanks.

**Comment by EmilAlipiev** [comment]
why are you inserting again another restriction? i dont understand. just let it to be not dependent. i have spent half a day to solve this problem because of this library. problem to problem is occurring with this library. 

```
Error	NU1107	Version conflict detected for Xamarin.Android.Support.Collections. Install/reference Xamarin.Android.Support.Collections 28.0.0.1 directly to project myApp.Droid.Shared to resolve this issue. 
 myApp.Droid.Shared -> Xamarin.Android.Support.v7.MediaRouter 28.0.0.1 -> Xamarin.Android.Support.Media.Compat 28.0.0.1 -> Xamarin.Android.Support.VersionedParcelable 28.0.0.1 -> Xamarin.Android.Support.Collections (= 28.0.0.1) 
 myApp.Droid.Shared -> Com.OneSignal 3.3.0-beta -> Xamarin.Android.Support.CustomTabs 28.0.0 -> Xamarin.Android.Support.Collections (= 28.0.0).	myApp.Droid.Shared	C:\Git\myApp\myApp\myApp.Droid.Shared\myApp.Droid.Shared.csproj	1	

```

**Comment by jabbourac** [comment]
> why are you inserting again another restriction? i dont understand. just let it to be not dependent. i have spent half a day to solve this problem because of this library. problem to problem is occurring with this library.
> 
> ```
> Error	NU1107	Version conflict detected for Xamarin.Android.Support.Collections. Install/reference Xamarin.Android.Support.Collections 28.0.0.1 directly to project myApp.Droid.Shared to resolve this issue. 
>  myApp.Droid.Shared -> Xamarin.Android.Support.v7.MediaRouter 28.0.0.1 -> Xamarin.Android.Support.Media.Compat 28.0.0.1 -> Xamarin.Android.Support.VersionedParcelable 28.0.0.1 -> Xamarin.Android.Support.Collections (= 28.0.0.1) 
>  myApp.Droid.Shared -> Com.OneSignal 3.3.0-beta -> Xamarin.Android.Support.CustomTabs 28.0.0 -> Xamarin.Android.Support.Collections (= 28.0.0).	myApp.Droid.Shared	C:\Git\myApp\myApp\myApp.Droid.Shared\myApp.Droid.Shared.csproj	1	
> ```

I am also experiencing this issue.

**Comment by MKielblock** [comment]
Also have the issue, will try manually adding one dependency at a time

**Comment by jabbourac** [comment]
@jkasten2 I am targeting MonoAndroid9.0 as well.

**Comment by ArgusMagnus** [comment]
@jkasten2 It seems to be working for me, thanks.
Yet the question remains why you placing an upper bound at all, this will result in issues again with the next version..

**Comment by MKielblock** [comment]
@ArgusMagnus could you maybe tell me what versions of Xamarin.Forms & OneSignal you are using as I only get it working with older versions of both

**Comment by ArgusMagnus** [comment]
@MKielblock: Latest stable of XF (3.4.0.1029999) and OneSignal 3.3.0-beta

**Comment by MKielblock** [comment]
@ArgusMagnus thx, will give it a try

**Comment by GaelSauzet** [comment]
There is case where <TargetFrameworkVersion></TargetFrameworkVersion> is not updated while set in the manifest and project option.
Updating that to v9.0 will then allow oneSignal v3.3.0-Beta to install.

**Comment by jabbourac** [comment]
I have gone through all the suggestions / troubleshooting steps mentioned in this thread, and I'm still experiencing the same dependencies issues with Custom Tabs / other Native Droid packages. I've lost confidence in this ever being fixed -- is anyone using an alternative library that's more reliable?

**Comment by kevinvella** [comment]
Hi,

For a while this issue was resolved using the the v3.3.0 beta.
But now it seems that you introduced a new constraint
Unable to resolve dependencies. 'Xamarin.Android.Support.v4 28.0.0.1' is not compatible with 'Com.OneSignal 3.3.0-beta constraint: Xamarin.Android.Support.v4 (>= 27.0.0 && < 27.2.0)'

I have project that is on v3.3.0 beta and 28.0.0.1
But i have another project that now cannot be updated to 28.0.0.1 because of this issue.

Any advise please?

**Comment by kevinvella** [comment]
Hi all,

Changed target framework version to android 9 and it works

<img width="401" alt="Screenshot 2019-03-09 at 16 41 36" src="https://user-images.githubusercontent.com/15779395/54073652-3db98800-428a-11e9-972a-52302432e881.png">
<img width="990" alt="Screenshot 2019-03-09 at 16 41 10" src="https://user-images.githubusercontent.com/15779395/54073653-3db98800-428a-11e9-9426-a78f58952933.png">
